### PR TITLE
ECO-3068: Implemented payment method specific models.

### PR DESCRIPTION
**Pull request by aleksey-kotsuba**
CR ticket - https://spryker.atlassian.net/browse/ECO-3069

### CODE REVIEW

File: src/SprykerEco/Yves/Heidelpay/Form/InvoiceSecuredB2cSubForm.php [hunk 1]
```diff
@@ -61,4 +75,68 @@ public function configureOptions(OptionsResolver $resolver): void
...
+    }
+
+    /**
+     * @return \Symfony\Component\Validator\Constraint
+     */
+    protected function createBirthdayConstraint(): Constraint
+    {
+        return new Callback([
+            'callback' => function ($date, ExecutionContextInterface $context) {
+                if (strtotime($date) > strtotime(self::MIN_BIRTHDAY_DATE_STRING)) {
```
Comment (gechetspr) [inline src/SprykerEco/Yves/Heidelpay/Form/InvoiceSecuredB2cSubForm.php]: Please change to static

Comment (gechetspr) [inline src/SprykerEco/Yves/Heidelpay/Form/InvoiceSecuredB2cSubForm.php]: Why not `>=`?
Also I would prefer to work with dates only via `DateTime` classes

File: src/SprykerEco/Zed/Heidelpay/Persistence/Propel/Schema/spy_heidelpay.schema.xml [hunk 2]
```diff
@@ -19,6 +19,9 @@
 
         <column name="easy_credit_fee" required="false" type="INTEGER" />
 
+        <column name="date_of_birth" required="false" type="VARCHAR"/>
```
Comment (gechetspr) [inline src/SprykerEco/Zed/Heidelpay/Persistence/Propel/Schema/spy_heidelpay.schema.xml]: IMO we should use date type here

File: src/SprykerEco/Yves/Heidelpay/Form/InvoiceSecuredB2cSubForm.php (hunk 1 above)
Comment (aleksey-kotsuba) [reply src/SprykerEco/Yves/Heidelpay/Form/InvoiceSecuredB2cSubForm.php]: Refactored to use `DateTime` object.
`>=` will affect customer that has turned 18 today.

**Pull request merged**